# HG changeset patch
# User Ben Abbott  <bpabbott@mac.com>
# Date 1371885840 -28800
#      Sat Jun 22 15:24:00 2013 +0800
# Node ID d3beeaf1d363e9a33a072d611a6688fcb98a1ce0
# Parent  56102c33122d36e13357bcf68cf8d2f0230bca97
Avoid changing the fontsize of a legend's text strings when printing.

legend.m: Changing the fontsize of a legend results in a legend's children
being replaced.  Rely upon the legend object font properties to in order to
update the legend's text strings' font properties.

diff --git a/scripts/plot/private/__gnuplot_print__.m b/scripts/plot/private/__gnuplot_print__.m
--- a/scripts/plot/private/__gnuplot_print__.m
+++ b/scripts/plot/private/__gnuplot_print__.m
@@ -288,16 +288,27 @@ function f = font_spec (opts, varargin)
     elseif (! isempty (opts.fontsize))
       f = sprintf ("fontsize %d", opts.fontsize);
     endif
   endswitch
 endfunction
 
 function [h, fontsize] = get_figure_text_objs (opts)
   h = findall (opts.figure, "-property", "fontsize");
+  hp = get (h, "parent");
+  if (iscell (hp))
+    hp = cell2mat (hp);
+  endif
+  ## Do not change the text objects fontsizes for the children of a
+  ## legend axes.  These will be handled by the fontsize listener.
+  is_legend_key_string = strcmp (get (hp, "tag"), "legend") ...
+                       & isprop (hp, "string") ...
+                       & isprop (hp, "location") ...
+                       & strcmp (get (hp, "type"), "axes");
+  h(is_legend_key_string) = [];
   fontsize = get (h, "fontsize");
   switch (numel (fontsize))
   case 0
     fontsize = {};
   case 1
     fontsize = {fontsize};
   endswitch
 endfunction
